### PR TITLE
[MM-10997] Add profile view on combined system message

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Add profile view on combined system message.

#### Ticket Link
Jira ticket: [MM-10997](https://mattermost.atlassian.net/browse/MM-10997)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes - see screen capture https://youtu.be/cN7MHHkRVhE
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates

### CODE REVIEW

File: components/post_view/combined_system_message/combined_system_message.jsx [hunk 1]
```diff
@@ -3,11 +3,13 @@
...
 import PropTypes from 'prop-types';
 import React from 'react';
-import {FormattedHTMLMessage, intlShape} from 'react-intl';
+import {intlShape} from 'react-intl';
 
-import {displayUsername} from 'mattermost-redux/utils/user_utils';
 import {Posts} from 'mattermost-redux/constants';
 
+import messageHtmlToComponent from 'utils/message_html_to_component';
+import {formatText} from 'utils/text_formatting.jsx';
```
Comment (saturninoabril) [inline components/post_view/combined_system_message/combined_system_message.jsx]: It seems convenient to use `Markdown` component but there's some issue on vertical spacing between group of system messages.

File: components/post_view/combined_system_message/combined_system_message.jsx [hunk 2]
```diff
@@ -191,117 +192,111 @@ export default class CombinedSystemMessage extends React.PureComponent {
...
-            );
+            localeHolder = {
+                id: postTypeMessage[postType].two.id,
+                defaultMessage: postTypeMessage[postType].two.defaultMessage,
+            };
         }
 
-        return formattedMessage;
+        const formattedMessage = formatMessage(localeHolder, {firstUser, secondUser, actor});
+        const formattedText = formatText(formattedMessage, options).replace(/p>/g, 'span>');
```
Comment (saturninoabril) [inline components/post_view/combined_system_message/combined_system_message.jsx]: This string replacement is an ugly hack to fix vertical spacing.  Basically changing `p` to `span` element.

**Comment by saturninoabril** [comment]
@esethna If you get a chance, may I request for a smoke test.  Thanks!

**Comment by esethna** [comment]
@lindalumitchell if we can please queue this for a QA pass

**Comment by lindy65** [comment]
Hey @saturninoabril @esethna 

I've tested on the spinmint and the feature is working as expected when clicking on the usernames in combined join/leave messages.  Looks good to me!

**Comment by saturninoabril** [comment]
Cool, thanks @lindy65!

File: components/post_view/combined_system_message/combined_system_message.jsx (hunk 2 above)
Comment (hmhealey) [reply components/post_view/combined_system_message/combined_system_message.jsx]: @saturninoabril If you use the Markdown component with the singleline option, does that fix the vertical spacing?

Comment (saturninoabril) [reply components/post_view/combined_system_message/combined_system_message.jsx]: I tried it first but had the same spacing issue.

Comment (saturninoabril) [reply components/post_view/combined_system_message/combined_system_message.jsx]: Oh wait, I missed the singleline option.  I'll find time to try it.

Comment (saturninoabril) [reply components/post_view/combined_system_message/combined_system_message.jsx]: Thanks @hmhealey!  Using Markdown with singleline option fixes the spacing.

**Pull request merged**